proposals/0653-amending-mrp.rst

```diff
@@ -0,0 +1,229 @@
+      ‘(>>)’ will eventually be removed in favour of ‘(*>)’
+      Remove definition for ‘(>>)’ (recommended) or define as ‘(>>) = (*>)’
+      See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/monad-of-no-return
+    |
+  8 |     (>>) _ b = b
+    |     ^^^^^^^^^^^^
+
+  example.hs:12:5: error: [-Wredundant-canonical-monad-instances]
+      ‘return’ definition detected
+      in the instance declaration for ‘Monad Id’.
+      ‘return’ will eventually be removed in favour of ‘pure’
+      Remove definition for ‘return’
+      See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/monad-of-no-return
+    |
+  12|     return = pure
+    |     ^^^^^^^^^^^^^
+
+  example.hs:13:5: error: [-Wredundant_canonical-monad-instances]
+      ‘(>>)’ definition detected
+      in the instance declaration for ‘Monad Id’.
+      ‘(>>)’ will eventually be removed in favour of ‘(*>)’
+      Remove definition for ‘(>>)’
+      See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/monad-of-no-return
+    |
+  13|     (>>) = (*>)
+    |     ^^^^^^^^^^^
+
+Effect and Interactions
+-----------------------
+Speeding up the phases of the proposal means that we will eventually be rid of
+the warnings and errors we are building up in service to this proposal.
+
+The alternative is that we warn against a change that we are not intending on 
+making.
+
+
+Costs and Drawbacks
+-------------------
+rhendric has already made a MR to perform phase 2 of the amended proposal, which
+can be found `here <https://gitlab.haskell.org/ghc/ghc/-/merge_requests/13999>`_.
+As such the only code cost would be for stage 3 of the amended proposal.
+
+Old tutorials may become more incorrect as they will no longer be able to define
+``Monad`` fully.
+
+Backward Compatibility
+----------------------
+This amendment would mean that we have breakage in uncommon cases. In one 
+attempt to compile Stackage with a patched GHC, ~94% of packages that were 
+attempted compiled. However, there are a handful of packages that do not want
+to accept fixes that would make them compatible with the changes in this 
+proposal which are depended upon by hundereds of other packages, which makes
+further analysis hard.
+
+However, the compiler error that is coming as a result of this proposal has been
+a warning since 2015, and has been on by default since 9.2.1 in 2020, so there
+shouldn't need to be more lead in.
+
+Alternatives
+------------
+Instead of amending the MRP proposal, we could try to do the proposal as 
+written.
+
+Unresolved Questions
+--------------------
+See `Proposed Library Change Specification` on the question of the eventual
+definition of ``(>>)``.
+
+As suggested by Teo on the discourse thread, we could put the breaking changes
+behind an on-by-default extension. However, I believe this is the incorrect
+move as we then have an increasingly complex combination of states to support,
+instead of cleaning up historical warts.
```

Comment:
```suggestion
As suggested by Teo on the Discourse thread, we could put the breaking changes
behind a language extension. This language extension would be added to the next GHCXXXX language edition. New code would therefore be disallowed from giving definitions of these methods, while old code would continue to compile. This comes with the disadvantage that we would have to keep the methods in the typeclass.

However, I believe this is the incorrect
move as we then have an increasingly complex combination of states to support,
instead of cleaning up historical warts.
```

Here's a bit more detail about what I had in mind.